compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SymbolEnter.java

```diff
@@ -1542,19 +1542,24 @@ private void addTopLevelNode(BLangPackage pkgNode, TopLevelNode node) {
 
     private void defineErrorDetails(List<BLangTypeDefinition> typeDefNodes, SymbolEnv pkgEnv) {
         for (BLangTypeDefinition typeDef : typeDefNodes) {
-            if (typeDef.typeNode.getKind() != NodeKind.ERROR_TYPE) {
-                continue;
+            BLangType typeNode = typeDef.typeNode;
+            if (typeNode.getKind() == NodeKind.ERROR_TYPE) {
+                SymbolEnv typeDefEnv = SymbolEnv.createTypeEnv(typeNode, typeDef.symbol.scope, pkgEnv);
+                BLangErrorType errorTypeNode = (BLangErrorType) typeNode;
+
+                BType detailType = Optional.ofNullable(errorTypeNode.detailType)
+                        .map(bLangType -> symResolver.resolveTypeNode(bLangType, typeDefEnv))
+                        .orElse(symTable.detailType);
+
+                ((BErrorType) typeDef.symbol.type).detailType = detailType;
+            } else if (typeNode.type != null && typeNode.type.tag == TypeTags.ERROR) {
+                SymbolEnv typeDefEnv = SymbolEnv.createTypeEnv(typeNode, typeDef.symbol.scope, pkgEnv);
+                BType detailType = ((BErrorType) typeNode.type).detailType;
+                if (detailType == symTable.noType) {
+                    BErrorType type = (BErrorType) symResolver.resolveTypeNode(typeNode, typeDefEnv);
+                    ((BErrorType) typeDef.symbol.type).detailType = type.detailType;
```

Comment:
Can the detail type here be null like for normal errors (what's handled in L1550-L1552)? If that is the case should we not set `symTable.detailType`?

Reply:
Also, is it possible for detail type to be `symTable.noType` when we get to L1576? Wouldn't we have set `symTable.detailType`?